src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -77,11 +81,16 @@ public CommandResult execute(Model model) throws CommandException {
         Person personToEdit = lastShownList.get(index.getZeroBased());
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+        if (personToEdit.equals(editedPerson)) {
```

Comment:
We use `Person.equals()` to check if any the fields of the editedPerson is different from the originalPerson. An error is thrown if they are the same, since there is nothing to update.

We use `Person.isSamePerson()` to check if any of the fields (Email, Phone, Telegram, Github) is the same as any of the other Persons in the list. An error is thrown if it returns true, since all these fields should be unique to a Person.
*`Person.isSamePerson()` has be renamed to `Person.checkDuplicateField()`.